### PR TITLE
feat(boa): adds normalize method

### LINKED ISSUE #13: Implement String prototype methods
## String prototype methods
These methods would be added to the String prototype object which is implemented here:
https://github.com/boa-dev/boa/blob/master/boa/src/builtins/string/mod.rs

Spec: https://tc39.es/ecma262/#sec-properties-of-the-string-prototype-object

**Implementation Example?**
See [charAt()](https://github.com/boa-dev/boa/blob/master/boa/src/builtins/string/mod.rs#L77-L119) and charCodeAt() for a good example

- [x] charAt()
- [x] codePointAt()
- [x] charCodeAt()
- [x] concat() 
- [x] endsWith()
- [x] includes()
- [x] indexOf()
- [x] lastIndexOf()
- [ ] localeCompare()
- [x] match()
- [x] matchAll()
- [x] normalize()
- [x] padEnd()
- [x] padStart()
- [x] repeat()
- [x] replace()
- [x] replaceAll()
- [x] search()
- [x] slice()
- [x] split()
- [x] startsWith()
- [x] substring()
- [ ] toLocaleLowerCase()
- [ ] toLocaleUpperCase()
- [x] toLowerCase()
- [x] toString()
- [x] toUpperCase()
- [x] trim()
- [x] trimEnd()
- [x] trimStart()
- [x] toString()
- [x] valueOf()
- [x] [@@iterator]()

### PR DESCRIPTION
**Pull request by neeldug**
This Pull Request progresses on #13.

It changes the following:

- adds string.prototype.normalize
- uses `unicode_normalization` crate

### CODE REVIEW

**Comment by neeldug** [comment]
### Test262 conformance changes:
  | Test result | master count | PR count | difference |
  | :---------: | :----------: | :------: | :--------: |
  | Total | 78,897 | 78,897 | 0 |
  | Passed | 26,974 | 26,990 | **+16** |
  | Ignored | 15,616 | 15,616 | 0 |
  | Failed | 36,307 | 36,291 | **-16** |
  | Panics | 0 | 0 | 0 |
  | Conformance | 34.19% | 34.21% | **+0.02%** |
  
  <details><summary><b>Fixed tests:</b></summary>
  
  ```
  test/built-ins/String/prototype/normalize/form-is-not-valid-throws.js [strict mode] (previously Failed)
  test/built-ins/String/prototype/normalize/form-is-not-valid-throws.js (previously Failed)
  test/built-ins/String/prototype/normalize/return-normalized-string-from-coerced-form.js [strict mode] (previously Failed)
  test/built-ins/String/prototype/normalize/return-normalized-string-from-coerced-form.js (previously Failed)
  test/built-ins/String/prototype/normalize/name.js [strict mode] (previously Failed)
  test/built-ins/String/prototype/normalize/name.js (previously Failed)
  test/built-ins/String/prototype/normalize/return-normalized-string.js [strict mode] (previously Failed)
  test/built-ins/String/prototype/normalize/return-normalized-string.js (previously Failed)
  test/built-ins/String/prototype/normalize/return-normalized-string-using-default-parameter.js [strict mode] (previously Failed)
  test/built-ins/String/prototype/normalize/return-normalized-string-using-default-parameter.js (previously Failed)
  test/built-ins/String/prototype/normalize/return-abrupt-from-form.js [strict mode] (previously Failed)
  test/built-ins/String/prototype/normalize/return-abrupt-from-form.js (previously Failed)
  test/built-ins/String/prototype/normalize/return-abrupt-from-this.js [strict mode] (previously Failed)
  test/built-ins/String/prototype/normalize/return-abrupt-from-this.js (previously Failed)
  test/built-ins/String/prototype/normalize/normalize.js [strict mode] (previously Failed)
  test/built-ins/String/prototype/normalize/normalize.js (previously Failed)
  ```
  </details>
  
  Previously 10 erroneously passed as the method didn't exist.

File: boa/src/builtins/string/mod.rs [hunk 1]
```diff
@@ -792,6 +794,10 @@ impl String {
         )))
     }
 
+    // pub(crate) fn replace_all(this: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+    //
+    // }
+
```
Comment (Razican) [inline boa/src/builtins/string/mod.rs:800]: It seems there are some leftover comments here.

File: boa/src/builtins/string/mod.rs [hunk 2]
```diff
@@ -1329,6 +1335,42 @@ impl String {
...
+    ///
+    /// The normalize() method normalizes a string into a form specified in the Unicode® Standard Annex #15
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-string.prototype.normalize
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize
+
```
Comment (Razican) [inline boa/src/builtins/string/mod.rs:1348]: I would remove this extra empty line

File: boa/src/builtins/string/mod.rs [hunk 3]
```diff
@@ -1329,6 +1335,42 @@ impl String {
...
+        let f_str;
+
+        let f = match form.is_undefined() {
+            true => "NFC",
+            false => {
+                f_str = form.to_string(context)?;
+                f_str.as_str()
+            }
+        };
```
Comment (Razican) [inline boa/src/builtins/string/mod.rs:1362]: It's a bit strange to match a boolean, right? I would just use if/else.

**Review by Razican — COMMENTED** [review]
Looks pretty good, give a look to my comments :)

File: boa/src/builtins/string/mod.rs (hunk 3 above)
Comment (neeldug) [reply boa/src/builtins/string/mod.rs:1362]: Yeah I had it as if/else before, but changed it to see how it'd change lifetimes if at all, but I agree, probably more clear with an if/else.

**Review by Razican — APPROVED** [review]
Looks good, thanks!

**Review by HalidOdat — APPROVED** [review]

**Pull request merged**